gprofiler/profilers/java.py

```diff
@@ -755,15 +756,31 @@ def _proc_exit_callback(self, tid: int, pid: int, exit_code: int):
     def _handle_kernel_messages(self, messages):
         for message in messages:
             _, _, text = message
-            entry = get_oom_entry(text)
-            if entry and entry.pid in self._profiled_pids:
-                logger.warning("Profiled Java process OOM", oom=json.dumps(entry._asdict()))
+            oom_entry = get_oom_entry(text)
+            if oom_entry and oom_entry.pid in self._profiled_pids:
+                logger.warning("Profiled Java process OOM", oom=json.dumps(oom_entry._asdict()))
                 self._disable_profiling()  # paranoia
+                continue
 
-            entry = get_signal_entry(text)
-            if entry and entry.pid in self._profiled_pids:
-                logger.warning("Profiled Java process signalled", signal=json.dumps(entry._asdict()))
+            signal_entry = get_signal_entry(text)
+            if signal_entry is not None and signal_entry.pid in self._profiled_pids:
+                logger.warning("Profiled Java process signalled", signal=json.dumps(signal_entry._asdict()))
                 self._disable_profiling()  # paranoia
+                continue
+
+            if self._java_safemode:
```

Comment:
if java safemode and profiling enabled

Reply:
This is on purpose actually, The system-wide things - I want to continue collecting them also after profiling was stopped (so I can understand whether it was an anomaly or not)

Reply:
I won't have profiled PIDs anymore, that's true